src/runtime_src/core/edge/user/shim.h

```diff
@@ -26,11 +26,9 @@
 #include <fstream>
 #include <map>
 #include <vector>
-
-// Forward declaration
-namespace xrt_core {
-    class bo_cache;
-}
+#include <mutex>
+#include <memory>
+#include "core/common/bo_cache.h"
```

Comment:
I do think we should stick with forward declaration.  It looks like the real problem was xrt::core vs xrt_core?

Reply:
> 
> 
> I do think we should stick with forward declaration. It looks like the real problem was xrt::core vs xrt_core? and missing

There are multiple issues. xrt::core vs xrt_core is one of them. The other one is caused by forward declaration and unique_ptr. We will hit a static_assert in unique_ptr.h after I include <memory>. The last issue is the missing sys/mman.h for mmap/munmap calls.